components/tenant-mgt/org.wso2.carbon.tenant.mgt/src/main/java/org/wso2/carbon/tenant/mgt/services/TenantMgtImpl.java

```diff
@@ -140,6 +140,37 @@ public Tenant getTenant(String tenantUniqueID) throws TenantMgtException {
         return tenant;
     }
 
+    public Tenant getTenantByDomain(String domain) throws TenantMgtException {
+
+        TenantManager tenantManager = TenantMgtServiceComponent.getTenantManager();
+        Tenant tenant;
+        try {
+            int tenantID = tenantManager.getTenantId(domain);
+            tenant = (Tenant) tenantManager.getTenant(tenantID);
+            if (tenant == null) {
+                throw new TenantManagementClientException(ERROR_CODE_DOMAIN_NOT_FOUND.getCode(),
+                        String.format(ERROR_CODE_DOMAIN_NOT_FOUND.getMessage(), domain));
+            }
+        } catch (org.wso2.carbon.user.api.UserStoreException e) {
+            throw new TenantManagementServerException("Error while getting the tenant.", e);
+        }
+        return tenant;
+    }
+
+    public boolean isDomainAvailable(String domain) throws TenantMgtException {
+
+        TenantManager tenantManager = TenantMgtServiceComponent.getTenantManager();
+        try {
+            int tenantID = tenantManager.getTenantId(domain);
+            if (tenantID == -1) { //-1 implies no domain exist.
+                return true;
+            }
+        } catch (org.wso2.carbon.user.api.UserStoreException e) {
+            throw new TenantManagementServerException("Error while getting the tenant.", e);
```

Comment:
What about adding domain to the message ?

Reply:
+1. Will add.

Reply:
Addressed